lib/api.js

```diff
@@ -21,13 +21,13 @@ const sharedWorkers = require('./plugin-support/shared-workers');
 
 function resolveModules(modules) {
 	return arrify(modules).map(name => {
-		const modulePath = resolveCwd.silent(name);
-
+		let [modulePath, args] = Array.isArray(name) ? name : [name, {}];
```

Comment:
`args` should be an array, not an object. It might be clearer to set the default during destructuring, so even `['path/to/module']` works:

```suggestion
		let [modulePath, args = []] = typeof name === 'string' ? [name] : name;
```

Reply:
Thanks! I appreciate your time and correction, but this is exactly where I'm confused. 

**Why I think it's an object**
From https://github.com/avajs/ava/issues/1666#issue-292221666, I am assuming the `config` looks like this 
```
"ava": {
  "require": [
    ["@babel/register", {
      "ignore": ["test/*"]
    }]
  ]
}
```
so when destructing `require`
```javascript
let [modulePath, args] = typeof name === 'string' ? [name] : name;
```
`modulePath` is `"@babel/register"`
`args` is `{
      "ignore": ["test/*"]
    }`, which is an object.

Please correct me if I am wrong.

Reply:
That happens to be the argument that module is expecting. Looking at this again I think we should do this:

```suggestion
		let [modulePath, ...args] = typeof name === 'string' ? [name] : name;
```

Note the `...` in front of `args`. If we're not passed a string, we assume an array. The first item must be the module path and any remaining arguments will be arguments provided when calling that module's default export.

Reply:
Oh, That makes sense, thank you, I will commit these suggestions and then work on the test part.